### PR TITLE
Implement upload method

### PR DESCRIPTION
**Pull request by boyntoni**

### CODE REVIEW

File: test/client.spec.js [hunk 1]
```diff
@@ -87,42 +87,4 @@ describe('Client', function() {
             expect(baseParams.api_key).to.equal(MOCK_KEY);
         });
     });
-
```
Comment (boyntoni) [inline test/client.spec.js]: Mocking the `request-promise` effort is unnecessarily tasking, and would require an additional library (`mockery`). Given this is simply a test to make sure this is called, running any of the demos should provide sufficient "coverage"

File: examples/upload.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const JWPlatformAPI = require('./jwplatform/api');
+
+const jwApiInstance = new JWPlatformAPI({
+    apiKey: process.env.API_KEY,
+    apiSecret: process.env.API_SECRET,
+});
+
+const filePath = './examples/test.mp4';
```
Comment (ksindi) [inline examples/upload.js]: Use `__dirname` to make sure you can execute script in different directores

**Comment by ksindi** [comment]
@boyntoni also make sure to show example in README

File: jwplatform/api.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+const resolve = require('path').resolve;
+const rp = require('request-promise');
 
 class JWPlatformAPI {
     constructor(opts = { timeout: 5000 }) {
         const { apiKey, apiSecret, timeout } = opts;
         this._client = new Client(apiKey, apiSecret, timeout);
 
         this._loadResources();
+        this.upload = this.upload.bind(this);
```
Comment (jnatalzia) [inline jwplatform/api.js]: prob don't need this bind

File: examples/upload.js (hunk 2 above)
Comment (boyntoni) [reply examples/upload.js]: This example is only designed to be used in one scenario, through the Docker image (detailed in  `README`). Use of `__dirname` here breaks this functionality. Added an example to `README.md` however.

**Comment by boyntoni** [comment]
@ksindi  Made all but one of changes, just want your review prior to releasing and publishing.

**Comment by ksindi** [comment]
Excellent job, @boyntoni!

**Pull request merged**